src/main/java/hudson/plugins/sshslaves/verifiers/TrileadVersionSupportManager.java

```
@@ -0,0 +1,70 @@
+package hudson.plugins.sshslaves.verifiers;
+
+import com.trilead.ssh2.signature.DSASHA1Verify;
+import com.trilead.ssh2.signature.RSASHA1Verify;
+import hudson.plugins.sshslaves.Messages;
+
+import java.io.IOException;
+
+/**
+ * @author Michael Clarke
+ * @since 1.18
+ */
+final class TrileadVersionSupportManager {
+
+    static TrileadVersionSupport getTrileadSupport() {
+        try {
+            Thread.currentThread().getContextClassLoader().loadClass("com.trilead.ssh2.signature.KeyAlgorithmManager");
+            return createaVersion9Instance();
+        } catch (ReflectiveOperationException e) {
```

Better to catch all exceptions as well